pkg/kv/kvserver/store_snapshot.go

```diff
@@ -549,15 +549,22 @@ func (kvSS *kvBatchSnapshotStrategy) Receive(
 			}
 
 			timingTag.start("sst")
+			verifyCheckSum := snapshotChecksumVerification.Get(&s.ClusterSettings().SV)
 			// All batch operations are guaranteed to be point key or range key puts.
 			for batchReader.Next() {
+				ek, err := batchReader.EngineKey()
```

Comment:
A bunch of func calls below (e.g. `PutInternalRangeKey`) will decode the `EngineKey` internally. I wonder if the APIs should accept `EngineKey` instead of the raw one. Probably not to be addressed in PR.